### ISSUE TITLE
Add server-side metrics

### ISSUE THREAD

**Issue by tyrannosaurus-becks** [approzium/approzium#73]
Probably through Prometheus, but need to add more requirements to this.

**Comment by tyrannosaurus-becks** [comment]
We discussed this further today and I captured more details. Using OpenTelemetry is preferable. Cyral is using OpenCensus which we believe was merged into OpenTelemetry.

Also, rather than integrating every one of these metrics with the code in `authenticator.go`, we should instead create a new implementation of the `pb.AuthenticationServer` interface that _wraps_ `authenticator.go` and captures as many metrics as possible there, to minimize code complexity.

**Metrics to capture**

- Number of requests
- Number of failure responses
- Length of time a request takes
- Length of time a request takes _to an outside system_, so if there's high latency but it originates at Postgres or Vault for example, it's possible to tell where the problem is
- Number of times a caller's claimed identity doesn't match their actual identity

**Additional notes**

- At some point we will likely want to tag metrics with metadata, like the database type. These tags should be strongly typed rather than free form because there's a performance hit for free form ones. We should probably open a separate, follow-on issue for those, once the specs for it are more clear.
